### PR TITLE
Add metrics definition link to tool README.md[skip ci] 

### LINKED ISSUE #2771: [DOC]Qualification and Profiling tools metrics explanations
## Report incorrect documentation

**Location of incorrect documentation**
https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/tools/README.md

**Describe the problems or issues found in the documentation**
I think it would be nice if all "Analysis metrics" have a more specific explanation like this:
numTasks: total tasks num of a job/stage
Duration: duration of a job/stage
duration_sum: sum of tasks duration 
duration_max: max value of tasks durations
duration_min: min value of tasks durations 
duration_avg: avg value of tasks durations
diskBytesSpilled_sum: The number of on-disk bytes spilled by tasks
executorCPUTime_sum: CPU Time the executor spends actually running tasks(including fetching shuffle data)
executorDeserializeCPUTime_sum: CPU Time taken on the executor to deserialize tasks
executorDeserializeTime_sum: Time taken on the executor to deserialize this task
executorRunTime_sum: Time the executor spends actually running tasks (including fetching shuffle data)
gettingResultTime_sum: The time that the driver spends fetching tasks results from workers.
input_bytesRead_sum: Bytes read from storage in this job/stage
input_recordsRead_sum: The number of read records in this job/stage
jvmGCTime_sum: Amount of time the JVM spent in garbage collection while executing tasks.
memoryBytesSpilled_sum: The number of in-memory bytes spilled by tasks.
output_bytesWritten_sum: Bytes written in storage in this job/stage
output_recordsWritten_sum: The number of written records in this job/stage
peakExecutionMemory_max: Peak memory used by internal data structures created during shuffles, aggregations and joins. The value of this accumulator should be approximately the sum of the peak sizes across all such data structures created in this task. For SQL jobs, this only tracks all unsafe operators and ExternalSort.
resultSerializationTime_sum: The time spent serializing tasks results on a executor before sending it back to the driver.
resultSize_max: The number of bytes tasks transmitted back to the driver as the TaskResult.
sr_fetchWaitTime_sum: The time spent on fetching data (local and remote)
sr_localBlocksFetched_sum: the number of blocks read locally
sr_localBytesRead_sum: the number of bytes read locally
sr_remoteBlocksFetched_sum: the number of blocks read remotely
sr_remoteBytesRead_sum: the number of bytes read remotely
sr_remoteBytesReadToDisk_sum: the number of bytes read from remote to local disk
sr_totalBytesRead_sum: Total shuffle bytes read, includes both data read locally and data read from remote executors
sw_bytesWritten_sum: the number of bytes written
sw_recordsWritten_sum: the number of records written
sw_writeTime_sum: the time spent on shuffle writing

### PR DESCRIPTION
**Pull request by nvliyuan**
Signed-off-by: Yuan Li yuali@nvidia.com

Closes #2771

Add metrics documentation to the tool README.md

### CODE REVIEW

File: tools/README.md [hunk 1]
```diff
@@ -430,6 +430,9 @@ Failed jobs:
 +-----+------+--------+---------------------------------------------------------------------------------------------------+
 ```
 
+### What does the metrics mean
```
Comment (tgravescs) [inline tools/README.md:433]:
```suggestion
### Metrics Definitions
```

File: tools/README.md [hunk 2]
```diff
@@ -430,6 +430,9 @@ Failed jobs:
...
+All the metrics definition can be found in [spark source code](https://github.com/apache/spark/blob/v3.1.1/core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala) or [spark webUI doc](https://spark.apache.org/docs/3.1.2/web-ui.html#content).
```
Comment (tgravescs) [inline tools/README.md:434]: I'd prefer not to point to source code.  I think all the task metrics are documented here:
https://spark.apache.org/docs/latest/monitoring.html#executor-task-metrics  and executor metrics here: https://spark.apache.org/docs/latest/monitoring.html#executor-metrics so can you remove the TaskMetrics.scala link and put these instead.  Also can you change the webui one to reference latest: https://spark.apache.org/docs/latest/web-ui.html#content

Comment (nvliyuan) [reply tools/README.md]: yes, and I update it.

File: tools/README.md [hunk 3]
```diff
@@ -430,6 +430,9 @@ Failed jobs:
 +-----+------+--------+---------------------------------------------------------------------------------------------------+
 ```
 
+### Metrics Definitions
+All the metrics definition can be found in [executor task metrics doc](https://spark.apache.org/docs/latest/monitoring.html#executor-task-metrics) / [executor metrics odc](https://spark.apache.org/docs/latest/monitoring.html#executor-metrics) or [spark webUI doc](https://spark.apache.org/docs/latest/web-ui.html#content).
```
Comment (tgravescs) [inline tools/README.md:434]: misspelled doc after executor metrics, Please capitalize spark in "spark webUI

Comment (nvliyuan) [reply tools/README.md]: Done.

**Review by tgravescs — DISMISSED** [review]

**Comment by tgravescs** [comment]
build

File: tools/README.md [hunk 4]
```diff
@@ -430,6 +430,9 @@ Failed jobs:
 +-----+------+--------+---------------------------------------------------------------------------------------------------+
 ```
 
+### Metrics Definitions
+All the metrics definition can be found in [executor task metrics doc](https://spark.apache.org/docs/latest/monitoring.html#executor-task-metrics) / [executor metrics doc](https://spark.apache.org/docs/latest/monitoring.html#executor-metrics) or [SPARK webUI doc](https://spark.apache.org/docs/latest/web-ui.html#content).
+
```
Comment (sameerz) [inline tools/README.md:435]:
```suggestion
All the metrics definitions can be found in the [executor task metrics doc](https://spark.apache.org/docs/latest/monitoring.html#executor-task-metrics) / [executor metrics doc](https://spark.apache.org/docs/latest/monitoring.html#executor-metrics) or the [SPARK webUI doc](https://spark.apache.org/docs/latest/web-ui.html#content).

```

**Review by sameerz — CHANGES_REQUESTED** [review]
We can add [skip ci] to the title for this PR.

**Comment by nvliyuan** [comment]
build

**Comment by revans2** [comment]
build

**Review by revans2 — APPROVED** [review]

File: tools/README.md (hunk 4 above)
Comment (nvliyuan) [reply tools/README.md]: done

**Review by sameerz — APPROVED** [review]

**Pull request merged**